fairscale/experimental/optim/dynamic_loss_scaler.py

```diff
@@ -0,0 +1,203 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All rights reserved.
+#
+# This source code is licensed under the BSD license found in the
+# LICENSE file in the root directory of this source tree.
+
+"""
+To prevent underflow or overflow of gradients, Dynamic Loss Scaler is used to
+dynamically scale up and down gradients by scaling the loss.
+"""
+
+
+from collections import defaultdict
+from enum import Enum
+
+
+import torch
+
+
+class OptState(Enum):
+    READY = 0
+    UNSCALED = 1
+    STEPPED = 2
+
+
+def _refresh_per_optimizer_state():
+    return OptState.READY
+
+
+class DynamicLossScaler(object):
```

Comment:
Shall we add a link to the file that was used as a reference for this implementation?

Reply:
+1, easier for the future us and for readers to get some context

Reply:
Do you mean a link about how to use the scaler? I can comment as "Use the DynamicLossScaler similar with GradScaler except it can be applied to cup device. https://pytorch.org/docs/stable/notes/amp_examples.html#amp-examples

Reply:
I meant the actual GradScaler class whose implementation mirrors this except for how we detect Nans.